x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/IndexLifecycleRunner.java

```diff
@@ -204,33 +205,45 @@ void onErrorMaybeRetryFailedStep(String policy, IndexMetadata indexMetadata) {
             int currentRetryAttempt = lifecycleState.getFailedStepRetryCount() == null ? 1 : 1 + lifecycleState.getFailedStepRetryCount();
             logger.info("policy [{}] for index [{}] on an error step due to a transitive error, moving back to the failed " +
                 "step [{}] for execution. retry attempt [{}]", policy, index, lifecycleState.getFailedStep(), currentRetryAttempt);
-            clusterService.submitStateUpdateTask("ilm-retry-failed-step", new ClusterStateUpdateTask() {
-                @Override
-                public ClusterState execute(ClusterState currentState) {
-                    return IndexLifecycleTransition.moveClusterStateToPreviouslyFailedStep(currentState, index,
-                        nowSupplier, stepRegistry, true);
-                }
+            clusterService.submitStateUpdateTask(
+                String.format(Locale.ROOT, "ilm-retry-failed-step {policy [%s], index [%s], failedStep [%s]}", policy, index,
+                    failedStep.getKey()),
+                new ClusterStateUpdateTask() {
 
-                @Override
-                public void onFailure(String source, Exception e) {
-                    logger.error(new ParameterizedMessage("retry execution of step [{}] for index [{}] failed",
-                        failedStep.getKey().getName(), index), e);
-                }
+                    @Override
+                    public TimeValue timeout() {
+                        // we can afford to drop these requests if they timeout as on the next {@link
+                        // IndexLifecycleRunner#runPeriodicStep} run the policy will still be in the ERROR step, as we haven't been able
+                        // to move it back into the failed step, so we'll try again
+                        return LifecycleSettings.LIFECYCLE_STEP_MASTER_TIMEOUT_SETTING.get(clusterService.state().metadata().settings());
```

Comment:
I think we can use the setting we already created to manipulate the ILM related master timeouts. What do you think @dakrone ?

Reply:
That sounds reasonable to me :+1:

Reply:
@dakrone cool, thanks for confirming, this is ready for review then 🙏🏻